File: Programmers/CodingTestPractice/DFS_BFS/43165.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int answer = 0;

void DFS(vector<int> numbers, int target, int sum, int count){
    if(numbers.size() == count){
        if(sum == target) answer++;
        return;
    }
    DFS(numbers, target, sum + numbers[count], count+1);
    DFS(numbers, target, sum - numbers[count], count+1);
}

int solution(vector<int> numbers, int target){
    DFS(numbers, target, 0, 0);
    return answer;
}

int main(){
    vector<int> inputData = {1,1,1,1,1};
    int target = 3;
    cout << solution(inputData, target) << '\n';
    return 0;
} 

File: Programmers/codingTest_Kit/DFS_BFS/43165.java
package Programmers.codingTest_Kit.DFS_BFS;

class Solution {
    int answer = 0;
    public void dfs(int n, int target, int idx, int result, int[] numbers){
        if(idx == n ){
            if(result == target){
                answer++;
            }
            return;
        } else {
            dfs(n, target, idx + 1, result + numbers[idx], numbers);
            dfs(n, target, idx + 1, result - numbers[idx], numbers);
        }
    }

    public int solution(int[] numbers, int target) {
        
        dfs(numbers.length, target, 0, 0, numbers);
        return answer;
    }
}
